src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -763,32 +783,41 @@ private static PackageSpec GetProjectJsonSpec(IMSBuildItem specItem)
             return result;
         }
 
-        private static PackageSpec GetBaseSpec(IMSBuildItem specItem, ProjectStyle projectStyle)
+        private static PackageSpec GetBaseSpec(IMSBuildItem specItem, ProjectStyle projectStyle, IEnumerable<IMSBuildItem> items)
         {
-            var frameworkInfo = GetFrameworks(specItem)
-                .Select(framework => new TargetFrameworkInformation()
-                {
-                    FrameworkName = framework
-                })
-                .ToList();
-
-            var spec = new PackageSpec(frameworkInfo);
+            var spec = new PackageSpec();
             spec.RestoreMetadata = projectStyle == ProjectStyle.PackagesConfig
                 ? new PackagesConfigProjectRestoreMetadata()
                 : new ProjectRestoreMetadata();
             spec.FilePath = specItem.GetProperty("ProjectPath");
             spec.RestoreMetadata.ProjectName = specItem.GetProperty("ProjectName");
 
+            if (projectStyle == ProjectStyle.DotnetCliTool || projectStyle == ProjectStyle.Unknown)
```

Comment:
`DotnetCliTool` is *not* the `dotnet tool` gesture actually. It's the old project tools and those have a specific property that represents the tfm to restore against. Those are special.

Reply:
ProjectStyle.Unknown is NuGet not enabled projects. 
When a project is not NuGet enabled, it can only be a legacy project, so reading the tfm the same we did in the old made sense to me. 

I can try to clean it up if you want me to, It was just what worked, and I didn't want to complicate it.